codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -172,7 +167,9 @@ public boolean visit(Http2Stream stream) {
                 }
             });
         } catch (Http2Exception e) {
-            PlatformDependent.throwException(e);
+            throw isStreamError(e) ? connectionError(INTERNAL_ERROR, e, "unexpected stream error") : e;
```

Comment:
No. The point of this is to catch visitor exceptions, and close the connection as result of them.

I should reduce the scope to just the listener loop. Is this what you are asking for?